SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -395,16 +397,15 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
 
 // Generate the n roots of the n-th degree orthogonal polynomial.
 // ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
-double poly_eval(int nroots, std::vector<double> roots, double x) {
+double poly_eval(std::vector<double> roots, double x) {
```

Comment:
* do not pass vectors by value, it will create an unnecessary copy
* always pass vectors by reference, unless you actually want to make a copy
* maintain const-correctness, if you don't intend to change the input, then mark it as `const`

Vectors are almost always accepted as `const std::vector<double> &roots`

Reply:
Done.